### PR TITLE
ci: migrate from jenkins to github actions

### PR DESCRIPTION
**Pull request by amannocci**
## Description

* Migrate everything from Jenkins to GitHub Actions except the release process.

## Motivation & Context

* Jenkins is deprecated

## Next steps

* Migrate the release process to GitHub Action.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    branches:
+      - main
+
+# Limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```
Comment (reakaleek) [inline .github/workflows/ci.yml:20]: FYI @v1v prefers to not cancel running builds on main.

See https://github.com/elastic/apm-aws-lambda/pull/370#discussion_r1090825824

File: .github/workflows/ci-extends.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    branches:
+      - main
+
+# Limit the access of the generated GITHUB_TOKEN
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```
Comment (reakaleek) [inline .github/workflows/ci-extends.yml:20]: https://github.com/elastic/apm-mutating-webhook/pull/76/files#r1170485293

File: .github/workflows/ci-extends.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+---
+
+name: ci-extends
+
+on:
+  push:
+    branches:
+      - main
+      - feature/github-actions
```
Comment (reakaleek) [inline .github/workflows/ci-extends.yml:9]: is this still needed?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
+---
+
+name: ci
+
+on:
+  push:
+    branches:
+      - main
+      - feature/github-actions
```
Comment (reakaleek) [inline .github/workflows/ci.yml:9]: is this still needed?

File: .github/workflows/ci-extends.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  pull_request:
+    branches:
+      - main
```
Comment (v1v) [inline .github/workflows/ci-extends.yml:12]: These event will fail on forked PRs when accessing later on the secrets, what's the main goal for this particular workflow?

**Review by v1v — COMMENTED** [review]
Will the `.github/workflows/opentelemetry.yml` be included in this PR? If so, the secrets might be needed too

File: .github/workflows/ci-extends.yml (hunk 5 above)
Comment (amannocci) [reply .github/workflows/ci-extends.yml:11]: The initial goal was to separate the workflow with and without secrets. We still need to validate that the docker image is buildable and pushable.

Comment (v1v) [reply .github/workflows/ci-extends.yml:11]: I'm not fully onboarded to use this workflow on PRs for vary reasons:

- it won't work for forked PRs.
- security concerns, infosec already pointed us that random code should be approved by Elasticians before interacting with any GH secrets.

On the other hand, can we run this only for the branches in the upstream? If so, what do you think a different workflow name something like `publish-docker`?

If needed we can enable the Support for PRs in a follow up, but I prefer to play safe in terms of using secrets

**Review by v1v — COMMENTED** [review]
Overall LGTM, though some comments regarding the secrets access on forked PRs

File: .github/workflows/ci-extends.yml (hunk 5 above)
Comment (amannocci) [reply .github/workflows/ci-extends.yml]: I split the workflow into two sub-steps. We will build the docker image within the `ci` workflow without secrets to validate that we can do so. On the other one `ci-extends`, we create and publish the docker image to the registry only on push events.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        with:
+          name: test-results
+          path: |
+            **/junit-*.xml
+  package:
+    name: Package
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Package & Push
```
Comment (v1v) [inline .github/workflows/ci.yml:93]: I cannot find any credentials within this workflow, will this push?

Comment (amannocci) [reply .github/workflows/ci.yml]: I updated the description according to the script's behavior.

**Note**: It creates a package only.

File: .github/workflows/ci-publish.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
+---
+
+name: ci-publish
```
Comment (v1v) [inline .github/workflows/ci-publish.yml:3]: Can this workflow name changed? Maybe `publish-docker`. I personally see `ci` for building and testing but in this particular case it generates and publish the docker images

**Review by v1v — APPROVED** [review]

**Comment by cachedout** [comment]
@eyalkoren Is there a chance we could get a review of this in the next few days? Thanks!

**Comment by cachedout** [comment]
Adding @marclop or @jackshirazi as reviewers at the suggestion of @felixbarny .

**Review by jackshirazi — APPROVED** [review]

**Review by marclop — APPROVED** [review]

**Pull request merged**